pkg/server/home_workspaces.go

```diff
@@ -327,145 +181,182 @@ func (h *homeWorkspaceHandler) ServeHTTP(rw http.ResponseWriter, req *http.Reque
+			logger.WithValues("cluster", homeClusterName, "user", effectiveUser.GetName()).Error(err, "error authorizing request")
+			responsewriters.Forbidden(ctx, attr, rw, req, authorization.WorkspaceAccessNotPermittedReason, homeWorkspaceCodecs)
+			return
+		}
+		if decision != authorizer.DecisionAllow {
+			responsewriters.Forbidden(ctx, attr, rw, req, authorization.WorkspaceAccessNotPermittedReason, homeWorkspaceCodecs)
 			return
 		}
 
-		foundLocally, retryAfterSeconds, err := h.searchForWorkspaceAndRBACInLocalInformers(lcluster.Name, workspaceType == HomeClusterWorkspaceType, effectiveUser.GetName())
-		if err != nil {
+		logger.Info("Creating home ThisWorkspace", "cluster", homeClusterName.String(), "user", effectiveUser.GetName())
+		this, err = h.kcpClusterClient.Cluster(homeClusterName).TenancyV1alpha1().ThisWorkspaces().Create(ctx, &tenancyv1alpha1.ThisWorkspace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: tenancyv1alpha1.ThisWorkspaceName,
+			},
+			Spec: tenancyv1alpha1.ThisWorkspaceSpec{
+				Type: tenancyv1alpha1.ClusterWorkspaceTypeReference{
+					Name: "home",
+					Path: "root",
+				},
+			},
+		}, metav1.CreateOptions{})
+		if err != nil && !kerrors.IsAlreadyExists(err) {
 			responsewriters.InternalError(rw, req, err)
 			return
 		}
-		if foundLocally {
-			logger.V(4).Info("found home workspace", "retryAfter", retryAfterSeconds)
-			if retryAfterSeconds > 0 {
-				// Return a 429 status asking the client to try again after the creationDelay
-				rw.Header().Set("Retry-After", fmt.Sprintf("%d", retryAfterSeconds))
-				http.Error(rw, "Creating the home workspace", http.StatusTooManyRequests)
-				return
-			}
+	} else {
+		this = objs[0].(*tenancyv1alpha1.ThisWorkspace)
+	}
 
-			if requestInfo.IsResourceRequest && requestInfo.APIGroup == tenancyv1alpha1.SchemeGroupVersion.Group && requestInfo.Resource == "clusterworkspaces" {
-				// If the call to searchForWorkspaceAndRBACInLocalInformers above returns foundLocally=true, and the
-				// request is for clusterworkspaces, there is a chance the tenancy APIBinding is not yet ready (or
-				// the cache is stale). If that is the case, return a retry-after to give the APIBinding cache a
-				// chance to catch up.
-				if ready, err := h.tenancyAPIBindingReady(lcluster.Name); err != nil {
-					responsewriters.InternalError(rw, req, err)
-				} else if !ready {
-					rw.Header().Set("Retry-After", fmt.Sprintf("%d", 1))
-					http.Error(rw, "Creating the home workspace", http.StatusTooManyRequests)
-					return
-				}
-			}
+	// here we have a ThisWorkspace. Create ClusterRoleBinding. Again: if this is pre-existing
+	// and it is not belonging to the current user, the user will get a 403 through normal authorization.
 
-			h.apiHandler.ServeHTTP(rw, req)
+	if this.Status.Phase == tenancyv1alpha1.ClusterWorkspacePhaseScheduling {
+		logger.Info("Creating home ClusterRoleBinding", "cluster", homeClusterName.String(), "user", effectiveUser.GetName(), "name", "workspace-admin")
```

Comment:
what if this fails? should this be done be a controller?

Reply:
then the (first) access to the home workspace fails from the point of the users. That's enough. The user will retry.